### ISSUE TITLE
Loss function in Stochastic "Hard" Attention

### ISSUE THREAD

**Issue by kouui** [sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning#14]
Great tutorial, thanks!
In the case of "Hard" attention, you mentioned in your tutorial that "it is not differentiable" so maybe this is why a new objective function `Ls` is proposed in the original paper. 
Do you know how to implement that new objective function `Ls`? Will it be an additional loss term in the loss function similar to those in VAE?

Thanks for your attention to this matter.

**Comment by sgrvinod** [comment]
Hi, thanks! Sorry for the late reply, I've been traveling.

My understanding of hard attention and VAEs is _very_ superficial, sadly. Thanks for the question - I will now finally get around to reading about this (i.e. the reparameterization of the sampling to make the network deterministic/differentiable). With some luck, I'll have a proper answer for you soon.
